src/main/java/org/opensearch/flowframework/util/ParseUtils.java

```diff
@@ -451,4 +451,25 @@ public static String removingBackslashesAndQuotesInArrayInJsonString(String inpu
         matcher.appendTail(result);
         return result.toString();
     }
+
+    /**
+     * Takes a String to json object map and converts this to a String to String map
+     * @param stringToObjectMap The string to object map to be transformed
+     * @return the transformed map
+     * @throws Exception for issues processing map
+     */
+    public static Map<String, String> convertStringToObjectMapToStringToStringMap(Map<String, Object> stringToObjectMap) throws Exception {
+        try (Jsonb jsonb = JsonbBuilder.create()) {
+            Map<String, String> stringToStringMap = new HashMap<>();
+            for (Map.Entry<String, Object> entry : stringToObjectMap.entrySet()) {
+                Object value = entry.getValue();
+                if (value instanceof String) {
+                    stringToStringMap.put(entry.getKey(), (String) value);
+                } else {
+                    stringToStringMap.put(entry.getKey(), jsonb.toJson(value));
```

Comment:
does this handle nulls? If not we should do a null check.

Reply:
if the request body is misconfigured then we return the error : `{"error":"IOException: template content invalid for specified Content-Type."` Tested with an empty `description` value in the interface input